File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var secondHighest = function(s) {
    let nums = s.replace(/\D/g, "").split("");
    if (nums.length === 0) return -1;
    let sortedNums = nums.sort((a,b) => b - a);   
    for (let i = 1; i < sortedNums.length; i++){
        if (sortedNums[i] !== sortedNums[0]) {
            return sortedNums[i];
        }
    }
    return -1;
};  

File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.py
class Solution:
    def secondHighest(self, s: str) -> int:
        # return set of numbers only
        nums = set(re.sub('[^\d]', '', s))
        # if nums is empty, return -1
        if len(nums) == 0:
            return -1
        # if only one number, return -1
        if len(nums) == 1:
            return -1
        else: 
            # remove highest value
            nums.remove(max(nums))
            # return second highest value
            return max(nums)